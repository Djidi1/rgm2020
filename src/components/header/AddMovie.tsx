import React, { FC } from 'react';
import { connect } from 'react-redux';
import { ErrorMessage, Form, Formik } from 'formik';
import { addMovie } from '../../store/actions';
import { dialogFields } from '../../helpers/constants';
import {
  Button, Dialog, Field, FormControl,
} from '../../uiKit';

export const addMovieValidator = (values: {title: string}): {title?: string} => {
  const errors: {title?: string} = {};
  if (!values.title) {
    errors.title = 'Required';
  }
  return errors;
};

type AddMovieProps = {
  showAddMovie: boolean;
  toggleShowAddMovie: () => void;
  addMovieData: (object) => { finally: (object) => void };
};

export const AddMovie: FC<AddMovieProps> = ({ showAddMovie, toggleShowAddMovie, addMovieData }) => {
  const initValues = {
    title: '',
    release_date: '',
    poster_path: '',
    genres: [],
    overview: '',
    runtime: '',
  };

  const addMovieOnSubmit = (values, { setSubmitting }) => {
    addMovieData(values).finally(() => { setSubmitting(false); });
  };

  return (
    <Dialog
      title="Add movie"
      show={showAddMovie}
      toggleShowDialog={toggleShowAddMovie}
    >
      <Formik
        enableReinitialize
        initialValues={initValues}
        validate={addMovieValidator}
        onSubmit={addMovieOnSubmit}
      >
        {({ isSubmitting }) => (
          <Form>
            {dialogFields.map((field, index) => (
              <FormControl key={field.label || index}>
                <Field
                  formik
                  type={field.type}
                  name={field.name}
                  label={field.label}
                  placeholder={field.placeholder}
                  options={field.options}
                />
                <ErrorMessage name={field.name} component="div" />
              </FormControl>
            ))}
            <Button withBorder type="reset">Reset</Button>
            <Button primary disabled={isSubmitting} type="submit">Submit</Button>
          </Form>
        )}
      </Formik>
    </Dialog>
  );
};

const mapDispatchToProps = {
  addMovieData: addMovie,
};

const mapStateToProps = ({ movie }) => ({
  movie: movie.movie,
  add_loading: movie.add_loading,
  error: movie.error,
});

export default connect(mapStateToProps, mapDispatchToProps)(AddMovie);
