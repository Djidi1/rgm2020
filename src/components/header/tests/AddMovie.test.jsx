import React from 'react';
import renderer from 'react-test-renderer';
import { AddMovie } from '../AddMovie';

describe('AddMovie', () => {
  beforeEach(() => {
    jest.resetModules();
  });

  it('is expected to render successfully', () => {
    const wrapper = renderer
      .create(<AddMovie />)
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
