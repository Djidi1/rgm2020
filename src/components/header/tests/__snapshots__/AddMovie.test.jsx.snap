// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMovie is expected to render successfully 1`] = `
<div
  className="Dialog__DialogModal-sc-3qcoit-0 cndQDb"
>
  <div
    className="Dialog__DialogWrapper-sc-3qcoit-1 hLvUCg"
  >
    <button
      className="Dialog__DialogCloseButton-sc-3qcoit-3 foHsOM"
    >
      X
    </button>
    <h1
      className="Dialog__DialogTitle-sc-3qcoit-2 kbzahH"
    >
      Add movie
    </h1>
    <div
      className="Dialog__DialogBody-sc-3qcoit-4 dyCeVH"
    >
      <form
        action="#"
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <div
          className="FormControl-sc-1epu8gc-0 jjIxXR"
        >
          <input
            className="Field-sc-1e9avp0-1 bUmsZj"
            name="id"
            onBlur={[Function]}
            onChange={[Function]}
            type="hidden"
          />
        </div>
        <div
          className="FormControl-sc-1epu8gc-0 jjIxXR"
        >
          <label
            className="Field__Label-sc-1e9avp0-0 fTufzV"
          >
            Title
          </label>
          <input
            className="Field-sc-1e9avp0-1 bUmsZj"
            name="title"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Title here"
            type="text"
            value=""
          />
        </div>
        <div
          className="FormControl-sc-1epu8gc-0 jjIxXR"
        >
          <label
            className="Field__Label-sc-1e9avp0-0 fTufzV"
          >
            Release date
          </label>
          <input
            className="Field-sc-1e9avp0-1 bUmsZj"
            name="release_date"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Select Date"
            type="date"
            value=""
          />
        </div>
        <div
          className="FormControl-sc-1epu8gc-0 jjIxXR"
        >
          <label
            className="Field__Label-sc-1e9avp0-0 fTufzV"
          >
            Movie URL
          </label>
          <input
            className="Field-sc-1e9avp0-1 bUmsZj"
            name="poster_path"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Movie URL here"
            type="text"
            value=""
          />
        </div>
        <div
          className="FormControl-sc-1epu8gc-0 jjIxXR"
        >
          <label
            className="Field__Label-sc-1e9avp0-0 fTufzV"
          >
            Genre
          </label>
          <select
            className="Field-sc-1e9avp0-1 bUmsZj"
            multiple={true}
            name="genres"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Genre here"
            value={Array []}
          >
            <option>
              Comedy
            </option>
            <option>
              Drama
            </option>
            <option>
              Romance
            </option>
          </select>
        </div>
        <div
          className="FormControl-sc-1epu8gc-0 jjIxXR"
        >
          <label
            className="Field__Label-sc-1e9avp0-0 fTufzV"
          >
            Overview
          </label>
          <input
            className="Field-sc-1e9avp0-1 bUmsZj"
            name="overview"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Overview here"
            type="text"
            value=""
          />
        </div>
        <div
          className="FormControl-sc-1epu8gc-0 jjIxXR"
        >
          <label
            className="Field__Label-sc-1e9avp0-0 fTufzV"
          >
            Runtime
          </label>
          <input
            className="Field-sc-1e9avp0-1 bUmsZj"
            name="runtime"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Runtime here"
            type="number"
            value=""
          />
        </div>
        <button
          className="Button-pmfev1-0 jeuPdg"
          type="reset"
        >
          Reset
        </button>
        <button
          className="Button-pmfev1-0 iHLxvC"
          disabled={false}
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</div>
`;
